Dockerfile

```diff
@@ -19,15 +35,25 @@ RUN go get github.com/google/gops
 #
 # Start a new stage from scratch
 #
-FROM alpine
+FROM alpine:3.7
+RUN adduser -u 1000 -D user
+COPY --from=proot /proot /home/user/.runrootless/runrootless-proot
+COPY --from=runc /runc /home/user/bin/runc
+
 RUN apk --no-cache add ca-certificates
-WORKDIR /app
-COPY --from=builder /root/src/pulsar-heartbeat /app/
+COPY --from=builder /root/src/pulsar-heartbeat /home/user
 
 # a kesque cert but it can overwritten by mounting the same path ca-bundle.crt
 COPY --from=builder /root/config/kesque-pulsar.cert /etc/ssl/certs/ca-bundle.crt
 
 # Copy debug tools
-COPY --from=builder /go/bin/gops /app/gops
-
-ENTRYPOINT ./pulsar-heartbeat ./runtime.yml
+COPY --from=builder /go/bin/gops /home/user/gops
+RUN mkdir /home/user/run
+RUN chown -R user:0 /home/user && chmod g=u /home/user
```

Comment:
@lhotari - this line adds 62 mb to the docker image. I know that OpenShift requires specific file permissions, but does the process actually write to all of the directories? If it does, we can use the chown argument to the copy command: `COPY --chown=user:0`. If it doesn't, then we can chown just the few directories/files that the process needs to write to.

For reference, I use the `dive` tool to quickly inspect docker images. Here is an output for `dive datastax/pulsar-heartbeat:1.0.4`:

<img width="652" alt="Screen Shot 2021-07-06 at 9 04 23 AM" src="https://user-images.githubusercontent.com/47911938/124614462-def48100-de39-11eb-946d-ee7313d12bbd.png">

Reply:
I created a follow-up PR #15 to address that separately.

Reply:
you can check with `dive lhotari/pulsar-heartbeat:rootlessv3` , that's the built image off PR #15 . The total image size is now 66MB.

Reply:
thanks for introducing [dive](https://github.com/wagoodman/dive), very useful tool.

Reply:
@lhotari - https://github.com/datastax/pulsar-heartbeat/pull/15 looks good to me. 

> thanks for introducing dive, very useful tool.

Glad you found it useful. I really like it when trying to optimize a docker image's size.